application/models/services/SurveyThemeConfiguration.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+use LimeSurvey\Models\Services\Exception\NotFoundException;
+use LimeSurvey\Models\Services\Exception\PermissionDeniedException;
+use Permission;
+use Template;
+use TemplateConfiguration;
+use TemplateManifest;
+
+class SurveyThemeConfiguration
+{
+    private Permission $permission;
+
+    public function __construct(
+        Permission $permission
+    ) {
+        $this->permission = $permission;
+    }
+
+    /**
+     * @param int $surveyId
+     * @param array $props properties (in options json string)
+     *
+     * @return void
+     * @throws PermissionDeniedException
+     */
+    public function updateThemeOption($surveyId, $props): void 
+    {
+        if (
```

Comment:
Let's assume that John Smith has global permission for survey updates, but is denied permission to update 123456. In that case if this is survey 123456, then we need to throw the exception.

As it is, your code will only throw an error if both are false. This misses the case when a global setting is overriden by a survey setting.

Reply:
Global permissions supercede survey permissions, not the other way around.

Reply:
@c-schmitz thanks for pointing that out. Then the code is good here. Marking the conversation as resolved